### PR TITLE
Remove dead BMA code.

### PR DESCRIPTION
**Pull request by Randy424**
regarding: https://github.com/stolostron/backlog/issues/25204

About this PR:
- It is an effort to clear the console owned documents of dead BMA code.
- Specifically, we are avoiding CIM and AI code and targeting our credential creation code, bma-cluster creation code, bma helper functions, doc strings and doc links.
- Code for identifying BMA credentials/clusters in our table and code streaming their resources from kube api **have not been** removed.

### CODE REVIEW

File: frontend/src/lib/doc-util.tsx [hunk 1]
```diff
@@ -19,14 +19,12 @@ export const DOC_LINKS = {
...
     CREATE_CONNECTION_OPENSTACK: `${DOC_BASE_PATH}/multicluster_engine/index#creating-a-credential-for-openstack`,
     CREATE_CONNECTION_VIRTUALIZATION: `${DOC_BASE_PATH}/multicluster_engine/index#creating-a-credential-for-virtualization`,
-    CREATE_CONNECTION_BAREMETAL: `${DOC_BASE_PATH}/multicluster_engine/index#credentials`,
     CREATE_CONNECTION_ANSIBLE: `${DOC_BASE_PATH}/multicluster_engine/index#creating-a-credential-for-ansible`,
     CREATE_CONNECTION_REDHATCLOUD: `${DOC_BASE_PATH}/multicluster_engine/index#creating-a-credential-for-openshift-cluster-manager`,
     CREATE_CONNECTION_PROXY: `${DOC_BASE_PATH}/multicluster_engine/index#cluster-overview`,
     CREATE_CONNECTION_PROXY_VMWARE: `${DOC_BASE_PATH}/multicluster_engine/index#proxy-vm`,
     CREATE_CONNECTION_PROXY_AWS: `${DOC_BASE_PATH}/multicluster_engine/index#proxy-aws`,
     CREATE_CONNECTION_PROXY_AZURE: `${DOC_BASE_PATH}/multicluster_engine/index#proxy-azure`,
-    CREATE_CONNECTION_PROXY_BAREMETAL: `${DOC_BASE_PATH}/multicluster_engine/index#proxy-bare`,
```
Comment (Randy424) [inline frontend/src/lib/doc-util.tsx:29]: FYI @cadawson 
BMA link removal.

Comment (Randy424) [reply frontend/src/lib/doc-util.tsx:29]: https://github.com/stolostron/console/pull/2063/files#diff-235b60184cc6834bb0c45e7a9e3a7e0731dd01d475e87028e43f06547f243808L22

This line as well.

**Comment by KevinFCormier** [comment]
@Randy424 Nice work! I think there are still some handlebars templates that are only used for bare metal and in CreateCluster.test.tsx there are some BMC related items that could be removed.

**Comment by Randy424** [comment]
@KevinFCormier I've made some updates based on your comments.

**Pull request merged**